src/hasher.rs

```diff
@@ -1,37 +1,38 @@
-use std::hash::{Hasher, BuildHasher};
-use std::collections::{HashMap, HashSet};
-use std::mem::MaybeUninit;
+use crate::gxhash::{platform::*, *};
 
 use rand::RngCore;
-
-use crate::gxhash::*;
-use crate::gxhash::platform::*;
+use std::{
+    collections::{HashMap, HashSet},
+    hash::{BuildHasher, Hasher},
+    mem::MaybeUninit,
+};
 
 /// A `Hasher` for hashing an arbitrary stream of bytes.
 /// # Features
 /// - The fastest [`Hasher`] of its class<sup>1</sup>, for all input sizes
 /// - Highly collision resitant
 /// - DOS resistance thanks to seed randomization when using [`GxHasher::default()`]
-/// 
+///
 /// *<sup>1</sup>There might me faster alternatives, such as `fxhash` for very small input sizes, but that usually have low quality properties.*
+#[derive(Clone, Debug)]
 pub struct GxHasher {
-    state: State
+    state: State,
 }
 
 impl GxHasher {
     #[inline]
     fn with_state(state: State) -> GxHasher {
-        GxHasher { state: state }
+        GxHasher { state }
     }
 }
 
 impl Default for GxHasher {
     /// Creates a new hasher with a empty seed.
-    /// 
+    ///
     /// # Warning ⚠️
-    /// Not using a seed may make your [`Hasher`] vulnerable to DOS attacks.  
+    /// Not using a seed may make your [`Hasher`] vulnerable to DOS attacks.
```

Comment:
Beware you lost the markdown line break

<img width="664" alt="image" src="https://github.com/ogxd/gxhash/assets/12692438/07af0c67-b7e4-45e5-bc47-e78a094f0142">

Reply:
Indeed. It is better practice to add an actual empty line. In this case what happened was that my editor swallowed the spaces (remove whitespace at end of lines when saving).

Most markdown syntax checkers will also complain about missing empty lines before and after headers. Would you be ok with switching to adding empty lines?

I.e.:
```rust
    /// Creates a new hasher with a empty seed.
    ///
    /// # Warning ⚠️
    ///
    /// Not using a seed may make your [`Hasher`] vulnerable to DOS attacks.
    ///
    /// It is recommended to use [`GxBuildHasher::default()`] for improved DOS resistance.
    ///
```

Reply:
An empty line produces a different result: instead of a line break, it goes for a new paragraph. 

I've quickly checked how they write documentation for popular rust crate and it seems I have taken a bad habit using newlines everywhere. So I am fine with stopping the whitespace at the end of the lines. In this example, it may be better to simply remove the whitespace so that there is no line break anymore. Both sentences belong to the same paragraph (so I approve your initial change)